test/governance/Governor.test.js

```diff
@@ -1,669 +1,666 @@
-const ERC721 = artifacts.require('$ERC721');
-const ERC1155 = artifacts.require('$ERC1155');
-const ERC1271WalletMock = artifacts.require('ERC1271WalletMock');
+const Governor = '$GovernorMock';
+const CallReceiver = 'CallReceiverMock';
+const ERC721 = '$ERC721';
+const ERC1155 = '$ERC1155';
+const ERC1271WalletMock = 'ERC1271WalletMock';
 
 const TOKENS = [
-  { Token: artifacts.require('$ERC20Votes'), mode: 'blocknumber' },
-  { Token: artifacts.require('$ERC20VotesTimestampMock'), mode: 'timestamp' },
-  { Token: artifacts.require('$ERC20VotesLegacyMock'), mode: 'blocknumber' },
+  { Token: '$ERC20Votes', mode: 'blocknumber' },
+  { Token: '$ERC20VotesTimestampMock', mode: 'timestamp' },
+  { Token: '$ERC20VotesLegacyMock', mode: 'blocknumber' },
 ];
 
-contract('Governor', function (accounts) {
-  const [owner, proposer, voter1, voter2, voter3, voter4] = accounts;
+const name = 'OZ-Governor';
+const version = '1';
+const tokenName = 'MockToken';
+const tokenSymbol = 'MTKN';
+const tokenSupply = ethers.parseEther('100');
+const votingDelay = 4n;
+const votingPeriod = 16n;
+const value = ethers.parseEther('1');
+
+async function deployToken(contractName) {
+  try {
+    return await ethers.deployContract(contractName, [tokenName, tokenSymbol, tokenName, version]);
+  } catch (error) {
+    if (error.message == 'incorrect number of arguments to constructor') {
+      // ERC20VotesLegacyMock has a different construction that uses version='1' by default.
+      return ethers.deployContract(contractName, [tokenName, tokenSymbol, tokenName]);
+    }
+    throw error;
+  }
+}
+
+async function tokenPartial(contractName, mode, { owner, receiver, voter1, voter2, voter3, voter4 }) {
+  const token = await deployToken(contractName, [tokenName, tokenSymbol, version]);
+  const mock = await ethers.deployContract(Governor, [
+    name, // name
+    votingDelay, // initialVotingDelay
+    votingPeriod, // initialVotingPeriod
+    0, // initialProposalThreshold
+    token, // tokenAddress
+    10, // quorumNumeratorValue
+  ]);
+  const helperTmp = new GovernorHelper(mock, mode);
+  await owner.sendTransaction({ to: mock, value });
+  await token.$_mint(owner, tokenSupply);
+  await helperTmp.connect(owner).delegate({ token: token, to: voter1, value: ethers.parseEther('10') });
+  await helperTmp.connect(owner).delegate({ token: token, to: voter2, value: ethers.parseEther('7') });
+  await helperTmp.connect(owner).delegate({ token: token, to: voter3, value: ethers.parseEther('5') });
+  await helperTmp.connect(owner).delegate({ token: token, to: voter4, value: ethers.parseEther('2') });
+  const helper = helperTmp.setProposal(
+    [
+      {
+        target: receiver.target,
+        data: receiver.interface.encodeFunctionData('mockFunction'),
+        value,
+      },
+    ],
+    '<proposal description>',
+  );
+  return {
+    token,
+    mock,
+    helper,
+    proposal: helper.currentProposal,
+  };
+}
+
+async function fixture() {
```

Comment:
I decided to put the fixture outside, otherwise, I see easily that we end up with multiple fixtures, losing the benefit of having just one state to load.

This way we load every variable once and reset to that snapshot.